src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -771,15 +773,17 @@ private Matches() {}
     return Match.of(Objects::nonNull);
   }
 
-  public static final Match<Territory> TerritoryIsWater = Match.of(Territory::isWater);
+  public static Match<Territory> territoryIsWater() {
+    return Match.of(Territory::isWater);
+  }
 
   /**
    * Returns a match indicating the specified territory is an island.
    */
   public static Match<Territory> territoryIsIsland() {
     return Match.of(t -> {
       final Collection<Territory> neighbors = t.getData().getMap().getNeighbors(t);
-      return neighbors.size() == 1 && TerritoryIsWater.match(neighbors.iterator().next());
+      return neighbors.size() == 1 && territoryIsWater().match(neighbors.iterator().next());
```

Comment:
Not related to this PR, but could it be the case that the name of this method is not 100% accurate? 
If I see this correctly this method returns false when an island is surrounded by 2 seazones, although it should logically return true.

Reply:
It looks like this method was added by @veqryn in 0d7200a to support scrambling when scrambling is restricted to originating from islands only.

My game rule history knowledge is not great, but IIRC, scrambling was introduced in WW2 Pacific.  Looking at the WW2 Global map, I don't see any islands that do not match this definition.  It may be that this definition of an island is specific to WW2 Global.  If you compare French Madagascar in WW2 Global to Madagascar in WW2 Classic, you'll see the former is surrounded by a single sea zone, while the latter is surrounded by four sea zones.

So, I guess the real question is, are there any maps where "Scramble From Island Only" is enabled, but the map maker drew a big enough island that it is surrounded by multiple sea zones?  A quick GitHub search of the triplea-maps organization shows that this property is only enabled in the following maps:

* ww2eur40.xml
* ww2eur40_veqryn_balanced.xml
* ww2pac40.xml
* ww2pac40_balanced_veqryn.xml
* ww2global40.xml
* pact_of_steel_2.xml

Other than Pact of Steel 2, those are all WW2 Global maps, so they should be ok.  As PoS is based off of Revised, it has Madagascar surrounded by four sea zones, so that could potentially be a problem.